### PR TITLE
luci-app-speedtest: add new packages

### PR DESCRIPTION
**Pull request by animegasan**
# LuCI for Speedtest

image

I feel this is very helpful for those of us who don't want the hassle of going to the terminal just to check the internet speed of OpenWrt devices. Especially for those who are remote accessing from outside the network.

### CODE REVIEW

**Comment by animegasan** [comment]
@systemcrash Is this correct?

Forgive me as I'm not familiar with github and thank you

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+				]),
+				E('tr', {'class': 'tr cbi-rowstyle-2'},[
+					E('td', {'class': 'td left', 'width': '50%'}, _('Upload Speed')),
+					E('td', {'class': 'td', 'id': 'upload'}, '-')
+				])
+			])
+		];
+		var body = E('div', {'class': 'cbi-map'}, [
+			E('h2', {'class': 'section-title'}, _('Speedtest')),
+			E('div', {'class': 'cbi-map-descr'}, _('Here you can perform a speedtest to measure the basic aspects of your network connection.')),
```
Comment (systemcrash) [inline applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:119]:
```js
			E('div', {'class': 'cbi-map-descr'}, _('Here you can perform a speed test to measure the basic aspects of your network connection.')),
```

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+							status.textContent = _('Finished');
+							running = false;
+						})
+						.catch(function(err) {
+							status.textContent = err;
+							running = false;
+						});
+					}
+				}
+			}, _('Run Speedtest'))
```
Comment (systemcrash) [inline applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:67]:
```js
			}, _('Start'))
```

File: applications/luci-app-speedtest/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# This is free software, licensed under the Apache License, Version 2.0
+#
+# Copyright (C) 2024 Hilman Maulana <hilman0.0maulana@gmail.com>
+
+include $(TOPDIR)/rules.mk
+
+LUCI_TITLE:=LuCI for speedtest
```
Comment (systemcrash) [inline applications/luci-app-speedtest/Makefile:7]: `LUCI_TITLE:=LuCI for speedtestcpp`

File: applications/luci-app-speedtest/Makefile [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+LUCI_DEPENDS:=+speedtestcpp
+LUCI_DESCRIPTION:=LuCI support for Speedtestcpp
```
Comment (systemcrash) [inline applications/luci-app-speedtest/Makefile:9]:
```diff
-LUCI_DESCRIPTION:=LuCI support for Speedtestcpp
```

**Review by systemcrash — CHANGES_REQUESTED** [review]

**Comment by systemcrash** [comment]
How does the app handle timeout? e.g. if there is no network?

File: applications/luci-app-speedtest/Makefile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+PKG_MAINTAINER:=Hilman Maulana <hilman0.0maulana@gmail.com>
+PKG_VERSION:=1.1
```
Comment (dannil) [inline applications/luci-app-speedtest/Makefile:12]: Is there a specific reason why the initial version isn't 1.0?

File: applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+	"luci-app-speedtest": {
+		"description": "Grant access to speedtest",
+		"write": {
+			"file": {
+				"/usr/bin/speedtest": [ "exec"]
```
Comment (dannil) [inline applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json:6]:
```suggestion
				"/usr/bin/speedtest": ["exec"]
```

**Comment by animegasan** [comment]
> How does the app handle timeout? e.g. if there is no network?

I forgot about that, and I plan to do this,

```js
.catch(function(err) {
    if (err === '{ "error": "unable to retrieve your ip info" }') {
        status.textContent = _('Error: Unable to retrieve IP information. Please check your internet connection.');
    } else {
        status.textContent = _('Error: Unable to perform speedtest. Please check your internet connection.');
    }
    running = false;
});
```
Because the response from speedtest if there is no internet 

```json
{
  "error": "unable to retrieve your ip info"
}
```

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js (hunk 1 above)
Comment (animegasan) [reply applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:119]: OK

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js (hunk 2 above)
Comment (animegasan) [reply applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:67]: OK

File: applications/luci-app-speedtest/Makefile (hunk 3 above)
Comment (animegasan) [reply applications/luci-app-speedtest/Makefile:7]: OK

File: applications/luci-app-speedtest/Makefile (hunk 5 above)
Comment (animegasan) [reply applications/luci-app-speedtest/Makefile:12]: I was careless, I'll fix it

File: applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json (hunk 6 above)
Comment (animegasan) [reply applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json:6]: OK, I thought I had left spaces when I wrote, but apparently not

**Pull request closed without merging**